docs/index.asciidoc

```diff
@@ -15,70 +15,99 @@ END - GENERATED VARIABLES, DO NOT EDIT!
 
 [id="plugins-{type}s-{plugin}"]
 
-=== Example filter plugin
+=== PILAR Filter Plugin
 
 include::{include_path}/plugin_header.asciidoc[]
 
 ==== Description
 
-Add plugin description here
+The PILAR plugin ports the https://dl.acm.org/doi/abs/10.1109/ICSE48619.2023.00077[PILAR parser] implementation for dynamic parsing of log events. The PILAR parser builds an n-gram dictionary of tokens in order to determine which tokens in a log event are considered dynamic and which are considered static. This enables the parser to dynamically extract out portions of the log without requring formal regular expressions.
+
+This plugin format performs especially well against logs that have a consistent format (eg. syslogs, Apache, MySQL) but unlike regex-based parsers like Grok and Dissect, this plugin can "learn" what parts of a log can be extracted out based on past log data that the plugin has analyzed.
+
+This plugin's core algorithm has been modified from the original PILAR algorithm in order to be operational at high load. Specific changes made include:
+
+  * Changing from a batch log processing algorithm to a stream-based log parsing solution
+  * Ensuring the n-gram dictionary can operate if Logstash runs on multiple workers
+  * Extensive testing at high scale to ensure that the parser adds minimal latency to the overall Logstash pipeline
```

Comment:
We can get rid of the word extensive here, it's fluff

Reply:
Change the second bullet point to be more general: `* Support for parallel parsing of logs across multiple workers`. - I think the n-gram dictionary part is too specific